compiler/rustc_expand/src/expand.rs

```diff
@@ -1278,7 +1326,7 @@ impl<'a, 'b> MutVisitor for InvocationCollector<'a, 'b> {
             &mut self.cx.current_expansion.dir_ownership,
             DirOwnership::UnownedViaBlock,
         );
-        noop_visit_block(block, self);
+        assign_id!(self, &mut block.id, || noop_visit_block(block, self));
```

Comment:
This shouldn't be necessary because blocks by themselves do not support attributes (https://github.com/rust-lang/rust/pull/87146#issuecomment-881890210).